### DISCUSSION TITLE
Address finder

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#730] [EXPERIMENTAL: Components, pages and patterns]
This contribution was made by @simonnebMOJ.

---

### Contribution purpose

The address finder is a  development of the existing 'Addresses' pattern in the GOV.UK Design System. https://design-system.service.gov.uk/patterns/addresses/

There are several MOJ services that require staff to search for and then select an address from a returned list - for example:

IN NOMIS:
Prisoners contact (visitors)
Prisoners next of kin

In DPS:
Incident Management Service

As we move more staff services out of NOMIS, we will need a standard way of searching for addresses in DPS services.

This contribution suggests we need two main features in the component - one for searching for an address, and another component  for displaying the list of returned addresses.

Technically, the desire is to use the OS Places API as the source of the addresses - so the designs should consider the information coming back from that service - see here: https://osdatahub.os.uk/docs/places/technicalSpecification

**Comment by jenthomson** [comment]
On the manage applications project we have been exploring the use of an address lookup in our alpha for adding addresses of PIN contacts to be added. These applications are submitted on paper and the services is the first step to digitise this where staff input the application content into a DPS interface.

In this design we use the type ahead component to allow the user to start typing an address and select the correct one. This is based on the way that [Royal Mail](https://www.royalmail.com/find-a-postcode) do address lookups. We have also provided a link to switch to input the address manually.

The address lookup has been well received by staff because they find prisoners often don't know someones address they may not provide the full address. As the applications are handwritten there may be issues withe legibility.

We have not yet picked an API to use or done any accessibility testing on this. We plan to do this before we go into a private beta.

**Address lookup** 
![1-address-lookup-default](https://github.com/user-attachments/assets/b5984c26-6b0b-4855-8e2b-61dc561353d1)
![2-address-lookup-active](https://github.com/user-attachments/assets/595031ab-b8e2-41ba-a209-b25583719a2c)
![3-address-lookup-selected](https://github.com/user-attachments/assets/d5ed9bb5-f362-459b-878f-95914a49c6f5)

**Enter address manually**

![4-enter-address-manually](https://github.com/user-attachments/assets/d728ff31-f1b4-4d22-ab7e-8292d92a8e57)
